flaskapp/views/ivrflow.py

```diff
@@ -420,17 +418,36 @@ def get_profile():
     else:
         return message
 
+
 # http://127.0.0.1:5000/new_user?username=testuser&&type=patient&&timezone=US/Pacific&&calltime=5:30:00&&phone=123-456-789
 def new_user():
     all_args = request.args.to_dict()
     rec1 = User.create(**all_args)
     rec1.save()
-    return {"success" : 200, "newuser" : model_to_dict(rec1)}
+    return {"success": 200, "newuser": model_to_dict(rec1)}
+
 
 def unsubscribe():
     user_info = request.args.to_dict()
     ph = user_info.get('phone')
-    del_row  = User.delete().where(User.phone == ph)
+    del_row = User.delete().where(User.phone == ph)
     if del_row > 0:
         return {"success": 200, "message": "user unsubscribed"}
-    return {"message": "user not found", "failed": 400}
\ No newline at end of file
+    return {"message": "user not found", "failed": 400}
+
+
+# getting set of from_ numbers from twilio call logs  that are less than 7 days old
+def get_from_phone_number_from_twilio_call_logs():
+    account_sid = os.environ['TWILIO_ACCOUNT_SID']
+    auth_token = os.environ['TWILIO_AUTH_TOKEN']
+    client = Client(account_sid, auth_token)
+
+    start = datetime.datetime.now()
+    calls = [call.from_ for call in client.calls.list() if
+             ((start.date() - call.start_time.date())
+              + (datetime.timedelta(days=0, hours=start.hour, minutes=start.minute, seconds=start.second)
+                 - datetime.timedelta(days=0, hours=call.start_time.time().hour, minutes=call.start_time.time().minute,
```

Comment:
Not exactly sure about this construction; You're trying to retrieve all phone numbers in past 7 days without taking into account sec, min, hours; only days are matter, aren't they? does the following (below) line of code solves the problem?:

```
 calls = [call.from_ for call in client.calls.list() if ((start.date() - call.start_time.date()).days < 7]
```

Reply:
We are taking into account days, hours, min and sec.
1. timedelta(days=1) = 1 day, 0:00:00
2. z= start.date() - call.start_time.date()
3. y = (datetime.timedelta(days=0, hours=start.hour, minutes=start.minute, seconds=start.second)
                 - datetime.timedelta(days=0, hours=call.start_time.time().hour, minutes=call.start_time.time().minute,
                                      seconds=call.start_time.time().second))
4  z+y

example :
z= 2021-12-02(start.date)  2021-11-03(call.start_time.date )
z= 29 days, 0:00:00
 y = datetime(0, 20, 24 ,5 )(Today hours,minutes and second ) - datetime( 0, 21, 57 ,7) (hours,minutes and second when call is made)
y= -1 day, 22:26:58

Required lag time = z+y 

Required lag time = 28 days, 22:26:58
comparing 
if (Required lag time <=timedelta(days=7) )

Reply:
calls = [call.from_ for call in client.calls.list() if ((start.date() - call.start_time.date()).days < 7] it also works

Reply:
done!